### PR TITLE
CONN-2207 adding nhinc-component-manager to exchange-management

### PR DESCRIPTION
**Pull request by trantang**
adding nhinc-component-manager to exchange-management
adding exchange-management to the soapui-project

### CODE REVIEW

**Comment by trantang** [comment]
retest this please

File: Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java [hunk 1]
```diff
@@ -118,35 +144,347 @@ public RefreshExchangeManagerResponseMessageType refreshExchangeManager(
...
     @Override
-    public GetExchangeInfoViewResponseMessageType getExchangeInfoView(GetExchangeInfoViewRequestMessageType arg0) {
-        return null;
+    public SimpleExchangeManagementResponseMessageType listOrganizations(ListOrganizationsRequestMessageType arg0) {
+        LOG.trace("listOrganizations--call");
+        SimpleExchangeManagementResponseMessageType response = newSimpleResponse();
+        String exchangeName = arg0.getExchangeName();
+
+        if (StringUtils.isNotBlank(exchangeName)) {
+            response.getOrganizationList().addAll(getExchangeManager().getAllOrganizationsBy(exchangeName));
```
Comment (mhpnguyen) [inline Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java]: Does this logic throw ExchangeManagerException?

File: Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java [hunk 2]
```diff
@@ -118,35 +144,347 @@ public RefreshExchangeManagerResponseMessageType refreshExchangeManager(
...
+    @Override
+    public ListEndpointUrlInfoByNhinTargetCommunitiesResponseMessageType listEndpointUrlInfoByNhinTargetCommunities(
+        ListEndpointUrlInfoByNhinTargetCommunitiesRequestMessageType arg0) {
+        LOG.trace("listEndpointUrlInfoByNhinTargetCommunities--call");
+        ListEndpointUrlInfoByNhinTargetCommunitiesResponseMessageType response = new ListEndpointUrlInfoByNhinTargetCommunitiesResponseMessageType();
+        NhinTargetCommunitiesType targets = arg0.getNhinTargetCommunities();
+        String serviceName = arg0.getServiceName();
+
+        if (null == targets || StringUtils.isBlank(serviceName)) {
+            return response;
```
Comment (mhpnguyen) [inline Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java]: Do we need to give some kind message if target or serviceName is blank

**Review by plobre — APPROVED** [review]

**Review by ptambellini01 — APPROVED** [review]

File: Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java (hunk 1 above)
Comment (trantang) [reply Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java]: no this does not throw exception

**Review by mhpnguyen — APPROVED** [review]

**Pull request merged**